ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/FilterStringColLikeStringScalar.java

```diff
@@ -58,12 +56,11 @@ protected List<CheckerFactory> getCheckerFactories() {
    * Accepts simple LIKE patterns like "abc%" and creates corresponding checkers.
    */
   private static class BeginCheckerFactory implements CheckerFactory {
-    private static final Pattern BEGIN_PATTERN = Pattern.compile("([^_%]+)%");
-
     public Checker tryCreate(String pattern) {
-      Matcher matcher = BEGIN_PATTERN.matcher(pattern);
-      if (matcher.matches()) {
-        return new BeginChecker(matcher.group(1));
+      CheckerPattern checkerPattern = new CheckerPattern(UDFLike.PatternType.BEGIN, pattern);
+      String matche = checkerPattern.check();
```

Comment:
can we transform CheckerPattern to enum and use it like
````
UDFLikePattern.BEGIN.apply(pattern)
````

Reply:
OK, that's how I fixed it.